2009-07-22  Jakub Jelinek  <jakub@redhat.com>

	* Makeconfig (ASFLAGS): Append $(sysdep-ASFLAGS).
	* sysdeps/i386/Makefile (sysdep-ASFLAGS): Add -U__i686.

Index: glibc-2.5-20061008T1257/Makeconfig
===================================================================
--- glibc-2.5-20061008T1257.orig/Makeconfig
+++ glibc-2.5-20061008T1257/Makeconfig
@@ -762,12 +762,12 @@ endif
 # The assembler can generate debug information too.
 ifndef ASFLAGS
 ifeq ($(have-cpp-asm-debuginfo),yes)
-ASFLAGS := $(filter -g%,$(CFLAGS))
+ASFLAGS = $(filter -g%,$(CFLAGS))
 else
-ASFLAGS :=
+ASFLAGS =
 endif
 endif
-ASFLAGS += $(ASFLAGS-config)
+ASFLAGS += $(ASFLAGS-config) $(sysdep-ASFLAGS)
 
 ifndef BUILD_CC
 BUILD_CC = $(CC)
Index: glibc-2.5-20061008T1257/sysdeps/i386/Makefile
===================================================================
--- glibc-2.5-20061008T1257.orig/sysdeps/i386/Makefile
+++ glibc-2.5-20061008T1257/sysdeps/i386/Makefile
@@ -2,6 +2,8 @@
 # Every i386 port in use uses gas syntax (I think).
 asm-CPPFLAGS += -DGAS_SYNTAX
 
+sysdep-ASFLAGS += -U__i686
+
 # The i386 `long double' is a distinct type we support.
 long-double-fcts = yes
 
